### ISSUE TITLE
[Feature] non-privileged user in Docker build

### ISSUE THREAD

**Issue by piramiday** [stashapp/stash#684]
**Is your feature request related to a problem? Please describe.**
at present, `stash` is running as root in the Docker container.

**Describe the solution you'd like**
ideally, a new non-privileged user should be created to execute `stash`, so that root permissions are dropped asap.

a non-root user for `stash` would agree with best practices and provide tighter security.
if possible, it would be great to be able to control both UID and GID of any created file.

**Comment by piramiday** [comment]
bump?

**Comment by virusperfect** [comment]
I would love to see that feature as well. Either by allowing the --user option or by having environment variables PUID and PGID.

**Comment by devster31** [comment]
Can be potentially solved by the following:
```yml
services:
  stash:
    command: stash --config /config/config.yml
    ...
    environment:
      - USER=user
      - HOME=/config
    ...
    user: 1000:1000
    ...
```

**Comment by jgramling17** [comment]
> Can be potentially solved by the following:
> 
> ```yaml
> services:
>   stash:
>     command: stash --config /config/config.yml
>     ...
>     environment:
>       - USER=user
>       - HOME=/config
>     ...
>     user: 1000:1000
>     ...
> ```

I used this and it worked for a lot of use cases. There are some things that don't work when stash uses/edits system files that are owned by root.

Would love to see native support for this.

**Comment by virusperfect** [comment]
I am using `hotio/stash` and it works great with the PUID and PGID variables.

**Comment by DogmaDragon** [comment]
> I am using `hotio/stash` and it works great with the PUID and PGID variables.

While that's an option, it's known to cause issues from time to time so use at your own risk.

**Comment by ikmckenz** [comment]
Added a $20 bounty to this issue on OpenCollective.

**Comment by WithoutPants** [comment]
$20 bounty assigned (txn number [725161](https://opencollective.com/stashapp/contributions/725161))

**Comment by jgramling17** [comment]
I believe this causes issues with any scrapers that use py_common.

I recieved this in the logs during an attempted scrape:
```
scrapeSceneURL: input: scrapeSceneURL could not unmarshal json from script output: EOF
could not unmarshal json from script output: EOF
[Scrape / ManyVids] You need to install the python modules mentioned in requirements.txt
[Scrape / ManyVids] If you have pip (normally installed with python), run this command in a terminal from the directory the scraper is located: pip install -r requirements.txt
```

If you try and run the `pip install -r requirements.txt` command like it suggests
you will receive this error:
`WARNING: The directory '/.cache/pip' or its parent directory is not owned or is not writable by the current user. The cache has been disabled. Check the permissions and owner of that directory. If executing pip with sudo, you should use sudo's -H flag.`

TL;DR I think stash can't create the '/.cache/pip' directory without being root

**Comment by Kausha3** [comment]
I’m picking this up with a narrow Docker-image change: add configurable PUID/PGID support, move the default config/home path out of /root, and drop privileges before running stash while keeping the official image behavior documented. I’ll keep it focused to the production image and compose example first.

**Comment by liangtovi-debug** [comment]
I’m taking a focused pass on this with a container-only scope:\n\n1. Run the production image as a non-root user by default.\n2. Add configurable  /  support for host ownership alignment.\n3. Move default config home from  to  for non-root compatibility.\n4. Keep backward compatibility by allowing explicit  runtime overrides.\n\nI’ll submit a PR with compose/docs updates and validation notes.

**Comment by liangtovi-debug** [comment]
Follow-up with corrected formatting (previous comment had shell-escaped formatting issues).

I’m taking a focused pass on this with a container-only scope:

1. Run the production image as a non-root user by default.
2. Add configurable `PUID` / `PGID` support for host ownership alignment.
3. Move default config home from `/root/.stash` to `/config` for non-root compatibility.
4. Keep backward compatibility by allowing explicit `--user` runtime overrides.

I’ll submit a PR with compose/docs updates and validation notes.

**Comment by liangtovi-debug** [comment]
Implemented in PR #6954: https://github.com/stashapp/stash/pull/6954

This adds non-root defaults in the production image with configurable `PUID`/`PGID`, updates compose to mount config at `/config`, and includes docs notes + validation commands.

**Comment by codeboost-tr** [comment]
I would like to work on this issue - running stash as non-privileged user in Docker.

Thank you!
